src/apps/app_webhook.erl

```diff
@@ -0,0 +1,126 @@
+-module(app_webhook).
+-export([start/1, new_block/2, new_transaction/2]).
+-include("ar.hrl").
+-include("ar_config.hrl").
+
+-define(NUMBER_OF_TRIES, 10).
+-define(WAIT_BETWEEN_TRIES, 30 * 1000).
+
+-define(BASE_HEADERS, [
+	{<<"content-type">>, <<"application/json">>}
+]).
+
+start(Configs) ->
+	adt_simple:start(?MODULE, Configs).
+
+new_block(Configs, Block) ->
+	call_webhooks(Block, Configs),
+	Configs.
+
+new_transaction(Configs, Tx) ->
+	call_webhooks(Tx, Configs),
+	Configs.
+
+call_webhooks(Entity, Configs) ->
+	lists:foreach(
+		fun(#config_webhook { events = Events } = Config) ->
+			ok = case lists:member(entity_type(Entity), Events) of
+				true -> try_call_webhook(Entity, Config);
+				false -> ok
+			end
+		end,
+		Configs
+	).
+
+try_call_webhook(Entity, Config) ->
+	try_call_webhook(Entity, Config, 0).
+
+try_call_webhook(Entity, Config, N) when N < ?NUMBER_OF_TRIES ->
+	#config_webhook{
+		url = Url,
+		headers = Headers,
+		include_data = IncludeData
+	} = Config,
+	try
+		{ok, {Scheme, _UserInfo, Host, Port, Path, Query}} = http_uri:parse(Url),
+		{ok, Client} = fusco:start(atom_to_list(Scheme) ++ "://" ++ binary_to_list(Host) ++ ":" ++ integer_to_list(Port), []),
+		{ok, Result} =
+			fusco:request(
+				Client,
+				<<Path/binary, Query/binary>>,
+				<<"POST">>,
+				?BASE_HEADERS ++ Headers,
+				to_json(Entity, IncludeData),
+				10000
+			),
+		case Result of
+			{{<<"200">>, _}, _, _, _, _} ->
+				ok = fusco:disconnect(Client),
+				ar:info([successful_webhook_call, {
+					{type, entity_type(Entity)},
+					{id, entity_id(Entity)},
+					{url, Url},
+					{headers, Headers},
+					{response, Result}
+				}]),
+				Url;
+			UnsuccessfulResult ->
```

Comment:
I think, it only makes sense to retry `{error, connection_closed | connect_timeout | timeout}` because in other cases it's very unlikely a retry would help. May be also `5**` status codes. At the very least, I would not retry on exception.

Reply:
Let's chat about it. Maybe this is good enough for now, and we can improve this later. I also think it would be nice to rework the whole queuing logic at some point so it cycles through the outgoing messages on failure rather than getting stuck on one

Reply:
Yes, the queueing logic could be improved.

What are the downsides of changing it so that it only retries connection and server errors now?